pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller.go

```diff
@@ -207,6 +209,7 @@ func (r *NSXServiceAccountReconciler) updateNSXServiceAccountStatus(ctx *context
 		obj = o.DeepCopy()
 		obj.Status.Phase = nsxvmwarecomv1alpha1.NSXServiceAccountPhaseFailed
 		obj.Status.Reason = fmt.Sprintf("Error: %v", *e)
+		obj.Status.Conditions = nsxserviceaccount.GenerateNSXServiceAccountConditions(obj.Status.Conditions, obj.Generation, metav1.ConditionFalse, fmt.Sprintf("Error: %v", *e))
```

Comment:
Better to put this error detail in message, and use some high-level causes in Reason . Reason should be enum. Reason is like the error type, message is like the error message. If it's hard to identify error type, let's use RealizationError as the only Reason enum.

Reply:
Use Message instead of Reason. Reason is not used not since we only have 2 status currently and it can be covered by Phase.

Reply:
According to https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition , "reason" is required. Since we only have two states, maybe use string enum such as "RealizationSuccess" "RealizationError".

Reply:
Fixed.